src/main/webapp/scripts/itinerary-overview.js

```diff
@@ -1,83 +1,167 @@
+let start;
+let waypoints;
+let end;
+
 document.addEventListener('DOMContentLoaded', function() {
     const elems = document.querySelectorAll('select');
     const instances = M.FormSelect.init(elems, undefined);
 });
 
 
-function initMap() {
-    const directionsRenderer = new google.maps.DirectionsRenderer();
-    const directionsService = new google.maps.DirectionsService();
-    const transitLayer = new google.maps.TransitLayer();
-
-    const map = new google.maps.Map(document.getElementById("map"), {
-        zoom: 10,
-        center: "25 Main Street, Belvedere Tiburon, CA"
-    });
-
-    transitLayer.setMap(map);
-    directionsRenderer.setMap(map);
+function displayMap() {
+    getTripData().then()
```

Comment:
Consider making `displayMap` `async` and refactoring this to `await getTripData()` instead